---
title: "Maom Lab - Research"
layout: textlay
excerpt: "Maom Lab -- Research"
sitemap: false
permalink: /software/
---

# Software


## Molecular modeling software
Large community software projects for molecular modeling that we use and contribute to in the lab

#### [Rosetta](https://www.rosettacommons.org/)
Rosetta is a platform for protein structure modeling and design. It is developed by dosens of labs and has a broad applications to sample and score protein conformations.

#### [UCSF DOCK](http://dock.compbio.ucsf.edu/)
DOCK3.7 is designed for large-scale virtual screening of chemical libraries from [ZINC15](https://zinc15.docking.org/)



## High-content screening tools
A collection of tools to analyze Morphological Profiling experiments, which use high-throughput microsopy as readout for drug screens.
#### [MPStats]({{ site.url }}{{ site.baseurl }}/MPStats)
An R package for single-cell pharmacology
* Methods for morphological profiling assay development and quality control
* Statistical modeling for high-throughput/high-content screens
* Chemoinformatic analysis for drug-repurposing

#### [MPLearn]({{ site.url }}{{ site.baseurl }}/MPLearn)
A python package for deep learning for microscopy and morphological profiling modeling
* Image restoration (denoising, deconvolution, virtual staining)
* Representation learning
* Weak-supervision and multiple instance learning

#### [mema]({{ site.url }}{{ site.baseurl }}/mema)
An R package to analyze high-throughput electrophysiology experiments including ex-vivo multi-electrode array and automated patch-clamp experimental setups



## Chemoinformatic tools
A set of R packages we have developed to for pharmacology analysis, drug-repurposing, hit-picking, and target-identification

#### [BayesPharma]({{ site.url }}{{ site.baseurl }}/BayesPharma)
An R package for modeling dose response experiments. It implements a principled Bayesian workflow building on [brms](https://paul-buerkner.github.io/brms/) and [Stan](https://mc-stan.org/).

#### [CalCEN]({{ site.url }}{{ site.baseurl }}/CalCEN)
Build and analyze co-expression networks for eukaryotic pathogens

#### [BioChemPantry]({{ site.url}}{{ site.baseurl }}/BioChemPantry)
Curate chemical and biological annotation datasets

#### [Zr]({{ site.url }}{{ site.baseurl }}/Zr)
Interact with the [https://zinc15.docking.org](https://zinc15.docking.org/), a webserver for commercially available compounds developed by the Irwin lab at UCSF.

### [dock_campaign_template]({{ site.url }}{{ site.baseurl }}/dock_campaign_template)
Prepare and manage a UCSF Dock campaign using an opinionated organization

#### [posteRa]({{ site.url }}{{ site.baseurl }}/posteRa)
Interact with [Manifold](https://postera.ai/manifold), a webserver developed for retrosynthesis prediction developed by PostERA.

#### [SEAR]({{ site.url }}{{ site.baseurl }}/SEAR)
Interact with the [https://sea16.docking.org](https://sea16.docking.org/), a webserver to compute ligand similarity ensemble analysis developed by the Shoichet lab at UCSF.

#### [Bethany]({{ site.url }}{{ site.baseurl }}/Bethay)
Build and analyze multiple sequence alignments, phylogentic trees, and ancetral gene-reconstructions
